README.md

```diff
@@ -54,13 +54,33 @@ above:
 
 1. Run `cargo clean` to eliminate any cached dependencies that were built against
 the non-MIR `libstd`.
-2. For a binary project, run `MIRI_SYSROOT=~/.xargo/HOST cargo +nightly miri` to
-build and run your project; for a binary or library, use `MIRI_SYSROOT=~/.xargo/HOST cargo +nightly miri test`
-to run all tests in your project through Miri.
+2. To run all tests in your project through, Miri, use
+`MIRI_SYSROOT=~/.xargo/HOST cargo +nightly miri test`.
+3. If you have a binary project, you can run it through Miri using
+`MIRI_SYSROOT=~/.xargo/HOST cargo +nightly miri`.
 
-If you forget to set `MIRI_SYSROOT`, be sure to run `cargo clean` again before
-correcting it. Otherwise you are likely to get "dependency was built against possibly
-newer std" errors.
+### Common Problems
+
+When modifying the above instructions, you may encounter a number of confusing compiler
+errors.
+
+#### "constant evaluation error: no mir for `<function>`"
+
+You may have forgotten to set `MIRI_SYSROOT` when calling `cargo miri test`, and
+your program called into `std` or `core`. Be sure to set `MIRI_SYSROOT=~/.xargo/HOST`.
+
+#### "found possibly newer version of crate `std` which `<dependency>` depends on"
+
+Your build directory may contain artifacts from an earlier build that did/did not
+have `MIRI_SYSROOT` set. Run `cargo clean` before switching from non-Miri to Miri
+builds and vice-versa.
+
+#### "found crate `std` compiled by an incompatible version of rustc"
+
+You may be running `cargo miri test` with a different compiler version than the one
+used to build the MIR-enabled `std`. Be sure to consistently use the same toolchain,
```

Comment:
Specifically here, perhaps I should say that the `rust-toolchain` file may make cargo/rustc commands use an unexpected compiler version?

Reply:
Let's just kill that file.  Do you want to do it here or should I do it in a separate PR?

Reply:
I added a commit that kills it.

Reply:
Right, but it served a purpose. That's why I wrote further up (seems to got lost) that it should be renamed and applied manually.

Basically, rename the file to `rust-version` or so, and then change both Travis and AppVeyor CI to read that other file instead of `rust-toolchain`. The README also needs a matching update.

Reply:
To be clear - `rust-version` does not have any meaning to rustup?

Reply:
Renamed to `rust-version` updated Travis and AppVeyor; added sentence to the README about manually using it.

Reply:
> To be clear - rust-version does not have any meaning to rustup?

Yeah I just made that filename up.